### PR TITLE
desktop fix omnibus

### PR DESCRIPTION
**Pull request by patosullivan**
fixes tlon-3383
fixes tlon-3382
fixes tlon-3328
fixes tlon-3401

unfortunately does not fix mark's text selection issue, I added more detail to the linear ticket for that one.

### CODE REVIEW

File: packages/ui/src/components/ChatMessage/ChatMessage.tsx [hunk 1]
```diff
@@ -173,6 +178,26 @@ const ChatMessage = ({
...
+        */}
+        {!showAuthor &&
+        !!post.deliveryStatus &&
+        post.deliveryStatus !== 'failed' ? (
+          <View position="absolute" right={12} top={8}>
+            <ChatMessageDeliveryStatus status={post.deliveryStatus} />
+          </View>
+        ) : null}
+
+        {!showAuthor && deliveryFailed ? (
```
Comment (arthyn) [inline packages/ui/src/components/ChatMessage/ChatMessage.tsx:193]: seems like now we only show it when `showAuthor` is false is that what we want?

**Review by arthyn — APPROVED** [review]
lgtm just a few questions

File: packages/app/navigation/linking.ts [hunk 2]
```diff
@@ -93,12 +93,12 @@ export const getDesktopLinkingConfig = (
               ChatList: '',
               GroupChannels: 'group/:groupId',
               DM: {
-                path: 'dm/:channelId/:selectedPostId?',
-                parse: parsePathParams('channelId', 'selectedPostId'),
+                path: 'dm/:channelId/:screen',
+                parse: parsePathParams('selectedPostId', 'channelId', 'screen'),
```
[the remark this answers is not in the document]
Comment (patosullivan) [reply packages/app/navigation/linking.ts:97]: It's not required, but we need it if we want to route to a particular post.

File: packages/app/navigation/utils.ts [hunk 3]
```diff
@@ -203,11 +203,20 @@ export function getDesktopChannelRoute(
...
+                ...(groupId
+                  ? {
+                      groupId,
+                      // channelId,
+                    }
+                  : undefined),
+                ...(selectedPostId ? { selectedPostId } : undefined),
+              },
+            }
+          : {}),
```
[the remark this answers is not in the document]
Comment (patosullivan) [reply packages/app/navigation/utils.ts]: simplified in latest commit

File: packages/app/navigation/linking.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (patosullivan) [reply packages/app/navigation/linking.ts]: Wait, yeah, we can just remove this. Pushing that up now.

File: packages/shared/src/store/useChannelPosts.ts [hunk 4]
```diff
@@ -127,12 +127,19 @@ export const useChannelPosts = (options: UseChanelPostsParams) => {
...
       firstPage,
-      _allPages,
+      allPages,
       _firstPageParam
     ): UseChannelPostsPageParams | undefined => {
-      if (!firstPage.canFetchNewerPosts) {
+
+      // if any page has reached the newest post, we can't fetch any newer posts
+      // page order for allPages should be newest -> oldest
+      // but apparently on channels with less than 50 posts, the order is reversed (on web only)
```
[the remark this answers is not in the document]
Comment (patosullivan) [reply packages/shared/src/store/useChannelPosts.ts:136]: No, `useChannelPosts` is adding the page in a different position in the array

File: packages/ui/src/components/ChatMessage/ChatMessage.tsx (hunk 1 above)
Comment (patosullivan) [reply packages/ui/src/components/ChatMessage/ChatMessage.tsx:193]: It still shows in `ChatAuthorRow` when `showAuthor` is true

File: packages/shared/src/store/useChannelPosts.ts (hunk 4 above)
[the remark this answers is not in the document]
Comment (arthyn) [reply packages/shared/src/store/useChannelPosts.ts:136]: ohhh

**Review by arthyn — APPROVED** [review]
lgtm!

**Pull request merged**